pkg/cdi/registry.go

```diff
@@ -122,15 +122,12 @@ var (
 // GetRegistry returns the CDI registry. If any options are given, those
 // are applied to the registry.
 func GetRegistry(options ...Option) Registry {
```

Comment:
> Does it make sense to simplify this to:
> 
> ```
> func GetRegistry(options ...Option) Registry {
> 	cache := GetDefaultCache()
> 	_ = cache.Configure(options...)
> 	return cache
> }
> ```

We can't do that without breaking the registry interface.

> 
> Note that this requires the removal of the
> 
> ```
> 	DeviceDB() RegistryDeviceDB
> 	SpecDB() RegistrySpecDB
> ```
> 
> from the `Registry` interface. I don't think our primary consumers use this though.

Sure, but IMO it would not be downstream friendly to remove or break interfaces without giving (potentially unknown) downstream users some slack. I would not alter the registry interface since we'll eventually remove it. And I wouldn't remove it without marking it first deprecated. I was planning to do the deprecation once we have switched to the package-level functions ourselves, but it can be done earlier, too.

> 
> @marquiz does make a valid point about the implied link between the registry and the default cache and at least documenting that. As soon as we mark `GetRegistry` as deprecated, we do away with these concerns.

It is mentioned in the updated documentation (although I noticed a typo there, which I now fixed).

Reply:
OK. Let's not break the interface now.